### PR TITLE
Km 4901 breadcrumb mvp

### PR DESCRIPTION
**Pull request by kiley-mitti**
## Brief Description

New feature: breadcrumb MVP. This is ready for review and testing!

## JIRA Link

[ASTRO-4901](https://rocketcom.atlassian.net/browse/ASTRO-4901)
[Astro-5080](https://rocketcom.atlassian.net/browse/ASTRO-5080)

## Motivation and Context

Adding new components based on use request!

## Issues and Limitations
Mark did some testing of this with SPA routers and I think it's going to work well. I did a little research and most routers do have a way to delegate their link down into the web component.

## Types of changes

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

## Checklist

- [x] This PR adds or removes a Storybook story.
- [x] I have added tests to cover my changes. - WIP
- [ ] Regressions are passing and/or failures are documented
- [x] Changes have been checked in evergreen browsers

### CODE REVIEW

**Comment by kiley-mitti** [comment]
oh! no I can totally do that. I thought we wanted a prop that could be set by devs. I'll get that in post haste. :)

File: packages/web-components/src/components/rux-breadcrumb/test/breadcrumb.spec.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            <rux-breadcrumb-item href="#">Second</rux-breadcrumb-item>
+            <rux-breadcrumb-item href="#">Third</rux-breadcrumb-item>
+        </rux-breadcrumb>
+        `
+        await page.setContent(template)
+        const el = await page.locator('rux-breadcrumb')
+        const breadcrumbItem = el.locator(
+            `rux-breadcrumb-item[data-id='first']`
+        )
+        await expect(breadcrumbItem).toHaveClass('hydrated')
```
Comment (markacianfrani) [inline packages/web-components/src/components/rux-breadcrumb/test/breadcrumb.spec.ts:16]: we can nix this. all of our tests run through a fixture that will await and fail if the components aren't hydrated.

File: packages/web-components/src/components/rux-breadcrumb/test/basic/index.html [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                    <rux-breadcrumb-item href="#"
+                        ><rux-icon size="20px" icon="border-clear"></rux-icon
+                        >Third</rux-breadcrumb-item
+                    >
+                    <rux-breadcrumb-item href="#"
+                        ><rux-icon icon="border-clear"></rux-icon>Forth<rux-icon
+                            icon="border-clear"
+                        ></rux-icon
+                    ></rux-breadcrumb-item>
+                    <rux-breadcrumb-item href="#"
```
Comment (markacianfrani) [inline packages/web-components/src/components/rux-breadcrumb/test/basic/index.html:51]: lets add a last breadcrumb that's not a link either here or a new section. we want to make sure we capture that style in the vrt.

File: packages/web-components/src/components/rux-breadcrumb/rux-breadcrumb-item/rux-breadcrumb-item.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            const fontSize = window
+                .getComputedStyle(link)
+                .getPropertyValue('font-size')
+            const childEls = slot.assignedNodes({}) as HTMLElement[]
+            //slotted content might just be text which would return nothing
+            if (childEls.length) {
+                //give rux-icons the correct size
+                childEls.map((el) => {
+                    if (el.nodeName === 'RUX-ICON')
+                        el.setAttribute('size', fontSize)
```
Comment (markacianfrani) [inline packages/web-components/src/components/rux-breadcrumb/rux-breadcrumb-item/rux-breadcrumb-item.tsx:56]: Screen Shot 2023-01-12 at 10 55 47 AM

should the indicator size also change/scale with font size?

File: packages/web-components/src/components/rux-breadcrumb/test/basic/index.html (hunk 2 above)
Comment (kiley-mitti) [reply packages/web-components/src/components/rux-breadcrumb/test/basic/index.html]: for sure. I've modified the VRT basic page so its a bit more organized and comprehensive

File: packages/web-components/src/components/rux-breadcrumb/rux-breadcrumb-item/rux-breadcrumb-item.tsx (hunk 3 above)
Comment (kiley-mitti) [reply packages/web-components/src/components/rux-breadcrumb/rux-breadcrumb-item/rux-breadcrumb-item.tsx:51]: I think that is. nice to have but probably bears more discussion. Looking at the way we would have to do that in the current iteration could be tricky (what if breadcrumb-items have different font sizes.. where do we decide to pull the size from?.. do we make them all one size or dependent on their preceding breadcrumb-item?)

If we DO want to do this for the MVP I think it's possible though. We just need a quick chat about it first.

Comment (markacianfrani) [reply packages/web-components/src/components/rux-breadcrumb/rux-breadcrumb-item/rux-breadcrumb-item.tsx:51]: for MVP I think we can table this for now

**Review by markacianfrani — APPROVED** [review]

**Pull request merged**